pacai/ui/topGraphics.py

```diff
@@ -0,0 +1,600 @@
+        elif (direction == 'North'):
+            endpoints = (90 + delta, 90 - delta)
+        elif (direction == 'South'):
+            endpoints = (270 + delta, 270 - delta)
+        else:
+            endpoints = (0 + delta, 0 - delta)
+
+        return endpoints
+
+    def movePacman(self, position, direction, image):
+        screenPosition = self.to_screen(position)
+        endpoints = self.getEndpoints(direction, position)
+        r = graphicsConstants.PACMAN_SCALE * self.gridSize
+        graphicsUtils.moveCircle(image[0], screenPosition, r, endpoints)
+        graphicsUtils.refresh()
+
+    def animatePacman(self, pacman, prevPacman, image):
+        if self.frameTime < 0:
+            print('Press any key to step forward, "q" to play')
+            keys = graphicsUtils.wait_for_keys()
+            if 'q' in keys:
+                self.frameTime = 0.1
+
+        if self.frameTime > 0.01 or self.frameTime < 0:
+            fx, fy = self.getPosition(prevPacman)
+            px, py = self.getPosition(pacman)
+            frames = 4.0
+            for i in range(1, int(frames) + 1):
+                pos = (
+                    px * i / frames + fx * (frames - i) / frames,
+                    py * i / frames + fy * (frames - i) / frames
+                )
+
+                self.movePacman(pos, self.getDirection(pacman), image)
+                graphicsUtils.refresh()
+                graphicsUtils.sleep(abs(self.frameTime) / frames)
+        else:
+            self.movePacman(self.getPosition(pacman), self.getDirection(pacman), image)
+        graphicsUtils.refresh()
+
+    def getGhostColor(self, ghost, ghostIndex):
+        if ghost.scaredTimer > 0:
+            return graphicsConstants.SCARED_COLOR
+        else:
+            return graphicsConstants.GHOST_COLORS[ghostIndex]
+
+    def drawGhost(self, ghost, agentIndex):
+        pos = self.getPosition(ghost)
+        dir = self.getDirection(ghost)
+        (screen_x, screen_y) = (self.to_screen(pos))
+        coords = []
+        for (x, y) in graphicsConstants.GHOST_SHAPE:
+            coords.append((
+                x * self.gridSize * graphicsConstants.GHOST_SIZE + screen_x,
+                y * self.gridSize * graphicsConstants.GHOST_SIZE + screen_y
+            ))
+
+        colour = self.getGhostColor(ghost, agentIndex)
+        body = graphicsUtils.polygon(coords, colour, filled = 1)
+
+        dx = 0
+        dy = 0
+        if dir == 'North':
+            dy = -0.2
+        if dir == 'South':
+            dy = 0.2
+        if dir == 'East':
+            dx = 0.2
+        if dir == 'West':
+            dx = -0.2
+
+        ghostSize = self.gridSize * graphicsConstants.GHOST_SIZE
+
+        coords = (screen_x + ghostSize * (-0.3 + dx / 1.5),
+                  screen_y - ghostSize * (0.3 - dy / 1.5))
+        leftEye = graphicsUtils.circle(coords, ghostSize * 0.2,
+                                       graphicsConstants.WHITE, graphicsConstants.WHITE)
```

Comment:
I can't remember what the exact syntax is for breaking a line, but make sure it is what we are implementing.